File: core4/leetcode/Arrays/P88/main.cpp
/******************************************************************************

                              88.Merge Sorted Arrays

*******************************************************************************/

#include <iostream>
#include <vector>

using namespace std;



class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
        int i = m-1;   // last valid element of nums1
        int j = n-1;   // last valid element of nums2
        int k = n+m-1; // last index of nums1
        
        while(j>=0){
            
            if (i >= 0 && nums1[i]>nums2[j]){
                nums1[k]= nums1[i];
                i--;

            }else{
                nums1[k]= nums2[j];
                j--;
            }
            
            k--;

        }
        

    }
};


int main()
{   
    
    Solution object;
    
    vector<int> nums1={1,2,3,0,0,0};
    int m=3;
    
    vector<int> nums2={2,5,6};
    int n=3;
    
    object.merge(nums1,m,nums2,n);
    
    
    cout << "Merged Array:" << endl;
    for(int i=0; i<nums1.size();i++){
        cout << nums1[i];
    }
    


    return 0;
}

File: core4/leetcode/Arrays/P88/main.py
class Solution:
    def merge(self,nums1,m,nums2,n):
        
        i=m-1
        j=n-1
        k=m+n-1
        
        while j>=0 :
            if i>=0 and nums1[i]>nums2[j]:
                nums1[k]=nums1[i]
                i-=1
            else:
                nums1[k]=nums2[j]
                j-=1
            
            k-=1



if __name__ == "__main__":
    nums1 = [1,2,3,0,0,0]
    m = 3
    
    nums2 = [2,5,6]
    n = len(nums2)
    
    object = Solution()
    
    object.merge(nums1,m,nums2,n)
    
    print("The Merged array is :",nums1)
    

    

    


